### ISSUE TITLE
Use this Kit for an audio skill?

### ISSUE THREAD

**Issue by smussani** [AreYouFreeBusy/AlexaSkillsKit.NET#18]
Hi,
Thank you so much for writing this kit.  It has helped me a lot in my learning. 
I would like to make a request.  Is there a way you can write a sample that uses this kit to create an audio streaming skill? I would like to write my own skill that uses Alexa's new features for audio streaming but need some guidance and sample code.
https://developer.amazon.com/blogs/post/Tx1DSINBM8LUNHY/New-Alexa-Skills-Kit-ASK-Feature-Audio-Streaming-in-Alexa-Skills

Really appreciate your help
Sajjad

**Comment by stefann42** [comment]
Hello Sajjad, unfortunately the library doesn't currently implement the ASK enhancements that were introduces for audio skills. Specifically, the `directives` node isn't currently parsed from the incoming request.

```
"directives": [
      {
        "type": "AudioPlayer.Play",
        "playBehavior": "string",
        "audioItem": {
          "stream": {
            "token": "string",
            "url": "string",
            "offsetInMilliseconds": 0
          }
        }
      }
    ],
```

I'll try to get to this task in the next revision of the library. Or feel free to dig in and contribute the code. I accept pull requests.

**Comment by jaddie** [comment]
I have made a rough implemention of this on my copy and am using it to stream youtube videos and play the audio on my Echo, I'll see if I can push it up shortly & make a pull request to get the conversation rolling on improvements etc as its rather rough I think.

**Comment by smussani** [comment]
Jaddie,
This is good news!!  Looking forward to see your code around this.

**Comment by smussani** [comment]
Thanks.. You mentioned that you have this working on your Echo.  Can you add the code on how you are calling these Directives from your Speechlet.cs file (your client)?

**Comment by rdlauer** [comment]
@jaddie do you have your YouTube project available in a public repo by any chance? I'd love to see how your customizations to this library work. I'm actually trying to do a very similar thing (play YouTube audio w/ echo). I have the underlying Web API working just fine (can search YouTube, grab audio of top hit, convert to mp3, save it to disk, and return the URL), but beyond that I'm struggling to get the skill working!

**Comment by jaddie** [comment]
I will get back to you asap

**Comment by smussani** [comment]
Waiting anxiously for your reply. Still waiting to make this work for audio streaming.
Thanks for your help

**Comment by tamhinsf** [comment]
I have a rather rudimentary (read: ugly) fork of it working that allows you to play a song, enqueue the next song in response to the "AudioPlayer.PlaybackNearlyFinished" intent, and go to the next track in reply to "Amazon.NextIntent".   It's not production quality, not complete, nor are the hooks to check/create an intent response where they should ideally be.   That said ...

Diffs here
https://github.com/AreYouFreeBusy/AlexaSkillsKit.NET/compare/master...tamhinsf:master

Repo here
https://github.com/tamhinsf/AlexaSkillsKit.NET

Sample project that uses it here in conjunction with the Microsoft Groove API (in the Azure4Alexa/Groove subfolder):
https://github.com/tamhinsf/azure4alexa

**Comment by dg-racing** [comment]
For those that are still needing this (I HAD to have this done!), I took what @tamhinsf did and finished up the implementation for all audio requests and responses. I'm sure it needs cleaned up and some things changed to be cleaner but I needed this done so I went ahead and did it. I've got a skill pending approval now by Amazon with it. 
For is here https://github.com/dg-racing/AlexaSkillsKit.NET

Let me know what you think. I'm hoping maybe it could be pull request worthy.
